modules/internal/ChapelRange.chpl

```diff
@@ -2484,12 +2484,14 @@ operator :(r: range(?), type t: range(?)) {
       const flwlen = myFollowThis.sizeAs(myFollowThis.intIdxType);
       if boundsChecking && this.hasLast() {
         // this check is for typechecking only
-        if isBoundedRange(this) {
-          if this.sizeAs(intIdxType) < flwlen then
-            HaltWrappers.boundsCheckHalt("zippered iteration over a range with too few indices");
-        } else
+        if !isBoundedRange(this) then
           assert(false, "hasFirst && hasLast do not imply isBoundedRange");
       }
+      if flwlen != 0 then
+        if boundsChecking then
```

Comment:
There could be a small performance benefit from swapping the two ifs above

Reply:
Good point.  What do you think about incorporating it into the previous conditional as well, like:

```chapel
if boundsChecking {
  if this.hasLast() { ... }
  if fwlen != 0 { ... }
}
```

Reply:
Yep, that sounds good.